libs/shared/interfaces/src/lib/domain/application.entity.ts

```diff
@@ -1,40 +1,4 @@
-import {
-  Application,
-  ApplicationAdvancedSettings,
-  Commit,
-  DeploymentHistoryApplication,
-  Instance,
-  Link,
-  Status,
-} from 'qovery-typescript-axios'
-import { LoadingStatus } from '../types/loading-status.type'
-import { ServiceRunningStatus } from './service-running-status.interface'
+import { ContainerApplicationEntity } from './container-application.entity'
+import { GitApplicationEntity } from './git-application.entity'
 
-export interface ApplicationEntity extends Application {
-  status?: Status
-  links?: {
-    loadingStatus: LoadingStatus
-    items?: Link[]
-  }
-  instances?: {
-    loadingStatus: LoadingStatus
-    items?: Instance[]
-  }
-  commits?: {
-    loadingStatus: LoadingStatus
-    items?: Commit[]
-  }
-  running_status?: ServiceRunningStatus
-  deployments?: {
-    loadingStatus: LoadingStatus
-    items: DeploymentHistoryApplication[]
-  }
-  advanced_settings?: {
-    loadingStatus: LoadingStatus
-    current_settings?: ApplicationAdvancedSettings
-  }
-  default_advanced_settings?: {
-    loadingStatus: LoadingStatus
-    default_settings?: ApplicationAdvancedSettings
-  }
-}
+export interface ApplicationEntity extends GitApplicationEntity, ContainerApplicationEntity {}
```

Comment:
This sounds strange to me. It can lead to confusion and bug, an ApplicationEntity will never be both GitApplicationEntity AND ContainerApplicationEntity, it's either one or the other, not both.
It's "dangerous" to do it because you make believe the compilation that all fields of one are present on the other, so the compilation won't say nothing but in runtime, you may ask a GitApplicationEntity for some values that are supposed to be only on ContainerApplicationEntity.
Here I think, what we should have is:
`export type ApplicationEntity = GitApplicationEntity | ContainerApplicationEntity`

The consequence of that is everytime you want to access something really specific to one or the other, you have to cast it, for example the getServiceType would become:
![CleanShot 2022-09-08 at 11 00 19@2x](https://user-images.githubusercontent.com/6163954/189081207-6a2e7770-556e-4e17-b1b4-76739f1feae3.jpg)

I know it is more verbose, and I know it's a big change, but I'm 100% convinced that it should be like this.
Completely open to talk about that anytime!